File: LeetCode 190- Reverse Bits/reverse_bits.py
class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        res = 0
        for i in range(32):
            bit = (n >> i) & 1
            res += (bit << (31-i))
        return res


if __name__ == "__main__":
    solution = Solution()
    num = 43261596
    print(solution.reverseBits(num))

# output: 964176192

File: LeetCode 190- Reverse Bits/reverse_bits.go
package main

import "fmt"

func reverseBits(num uint32) uint32 {
	var res uint32 = 0
	for i := 0; i < 32; i++ {
		bit := (num >> i) & 1
		res += (bit << (31 - i))
	}
	return res
}

func main() {
	var num uint32 = 43261596
	fmt.Println(reverseBits(num))
}

// output: 964176192
